File: schemas/recipe-cooking/mongodb/meal_plan_entries.js
// meal_plan_entries: Individual recipe slots within a meal plan.
// See README.md for full design rationale.

const mongoose = require("mongoose");

const mealPlanEntrySchema = new mongoose.Schema(
  {
    meal_plan_id: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "MealPlan",
      required: true,
    },
    recipe_id: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Recipe",
      required: true,
    },
    plan_date: { type: String, required: true },
    meal_type: {
      type: String,
      enum: ["breakfast", "lunch", "dinner", "snack"],
      required: true,
    },
    servings: { type: Number, default: null },
    note: { type: String, default: null },
  },
  {
    timestamps: false,
  }
);

mealPlanEntrySchema.index({ meal_plan_id: 1, plan_date: 1 });
mealPlanEntrySchema.index({ recipe_id: 1 });

module.exports = mongoose.model("MealPlanEntry", mealPlanEntrySchema);

File: schemas/recipe-cooking/spacetimedb/meal_plan_entries.rs
// meal_plan_entries: Individual meal slots within a meal plan.
// See README.md for full design rationale.

// Composite index(meal_plan_id, plan_date) — not supported inline, enforce in reducer logic

#[derive(SpacetimeType, Clone)]
pub enum MealType {
    Breakfast, // type: String
    Lunch,
    Dinner,
    Snack,
}

#[spacetimedb::table(name = meal_plan_entries, public)]
pub struct MealPlanEntry {
    #[primary_key]
    pub id: String, // UUID

    #[index(btree)]
    pub meal_plan_id: String, // FK → meal_plans.id (cascade delete)

    #[index(btree)]
    pub recipe_id: String, // FK → recipes.id (cascade delete)

    pub plan_date: String, // YYYY-MM-DD
    pub meal_type: MealType,
    pub servings: Option<i32>,
    pub note: Option<String>,
}
